controller/share_manager_controller.go

```diff
@@ -299,6 +299,12 @@ func (c *ShareManagerController) syncShareManager(key string) (err error) {
 		if err := c.cleanupShareManagerPod(sm); err != nil {
 			return err
 		}
+
+		if err := c.ds.DeleteConfigMap(c.namespace, types.GetConfigMapNameNameFromShareManagerName(sm.Name)); err != nil {
+			log.Warnf("Failed to delete the configmap (recovery backend) for share manager %v since %v",
```

Comment:
If can't delete, it will be left over? Should it be a transaction together by returning the error?

Reply:
I think we need to return error here for avoiding the left-over.